tests/suites/test_suite_ssl.function

```diff
@@ -928,31 +928,86 @@ int mbedtls_move_handshake_to_state( mbedtls_ssl_context *ssl,
 #endif /* MBEDTLS_X509_CRT_PARSE_C */
 
 /*
- * Write application data. Then increase write and fragments counter
+ * Write application data. Increase write counter and fragments counter if
+ * necessary.
  */
 int mbedtls_ssl_write_fragment( mbedtls_ssl_context *ssl, unsigned char *buf,
```

Comment:
Also, I would suggest getting rid of this function whatsoever. You're doing checks here that are duplicated in the places that call this function, furthermore, `expected_frames == 1` and `expected_frames>1` can be done in a more elegant way (see below).

Reply:
This function was added according to @Patater suggestions from other PR and now this function has been extended to include checks that otherwise would be duplicated for the client and server.